src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClients/FailOverSupportedConfigurationClient.cs

```diff
@@ -0,0 +1,253 @@
+            this._parallelRetryInterval = options.ParallelRetryInterval;
+        }
+
+        public FailOverSupportedConfigurationClient(IEnumerable<Uri> endpoints, TokenCredential credential, AzureAppConfigurationOptions options)
+        {
+            if (endpoints == null || endpoints.Count() < 1)
+            {
+                throw new ArgumentNullException(nameof(endpoints));
+            }
+
+            if (credential == null)
+            {
+                throw new NullReferenceException(nameof(credential));
+            }
+
+            this._configurationClientAndStates = endpoints.Select(endpoint => (new ConfigurationClient(endpoint, credential, options.ClientOptions), new ConfigurationClientState(endpoint)));
+            this._parallelRetryInterval = options.ParallelRetryInterval;
+        }
+
+        public Task<Response<ConfigurationSetting>> GetConfigurationSettingAsync(string key, string label = null, CancellationToken cancellationToken = default)
+        {
+            Task<Response<ConfigurationSetting>> func(ConfigurationClient client)
+            {
+                return client.GetConfigurationSettingAsync(key, label, cancellationToken);
+            }
+
+            return ExecuteWithFailOverPolicyAsync(func, forceTryFailedReplicas: false, cancellationToken);
+        }
+
+        public Task<Response<ConfigurationSetting>> GetConfigurationSettingAsync(ConfigurationSetting setting, bool onlyIfChanged = false, CancellationToken cancellationToken = default)
+        {
+            Task<Response<ConfigurationSetting>> func(ConfigurationClient client)
+            {
+                return client.GetConfigurationSettingAsync(setting, onlyIfChanged, cancellationToken);
+            }
+
+            return ExecuteWithFailOverPolicyAsync(func, forceTryFailedReplicas: false, cancellationToken);
+        }
+
+        public async Task<IEnumerable<ConfigurationSetting>> GetConfigurationSettingsAsync(SettingSelector selector, CancellationToken cancellationToken = default)
+        {
+            Page<ConfigurationSetting> settingsPage = null;
+            string continuationToken = null;
+            IAsyncEnumerator<Page<ConfigurationSetting>> enumerator;
+            var result = new List<ConfigurationSetting>();
+
+            IAsyncEnumerator<Page<ConfigurationSetting>> func(ConfigurationClient client)
+            {
+                return client.GetConfigurationSettingsAsync(selector, cancellationToken)
+                             .AsPages(continuationToken: continuationToken)
+                             .GetAsyncEnumerator();
+            }
+
+            do
+            {
+                enumerator = await ExecuteWithFailOverPolicyAsync(func, forceTryFailedReplicas: false, cancellationToken);
+                settingsPage = enumerator != null && await enumerator.MoveNextAsync() ? enumerator.Current : null;
```

Comment:
`MoveNextAsync` is the piece making teh call to AAC right? That should be using the failover policy.

Reply:
Even though `MoveNextAsync` is the piece making the call to AAC, which `ConfigurationClient` to use is defined by the `enumerator`.  That's the reason we need to get the `enumerator` with failover policy and execute `MoveNextAsync`.

Reply:
All of the async calls need to be in the method we pass to the execution policy, so perhaps we just need to move MoveNextAsync in there.